compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEBufferizeToAllocation.cpp

```diff
@@ -159,20 +159,23 @@ void AMDAIEBufferizeToAllocationPass::runOnOperation() {
   mlir::FunctionOpInterface funcOp = getOperation();
   linalg::LinalgOp linalgOp;
   funcOp->walk<WalkOrder::PostOrder, ReverseIterator>([&](linalg::LinalgOp op) {
+    if (!isElementwise(op) && !linalg::isaContractionOpInterface(op) &&
+        !linalg::isaConvolutionOpInterface(op)) {
+      return WalkResult::advance();
+    }
+    if (isa<linalg::FillOp>(op)) {
```

Comment:
I think this should be redundant as the first if statement should take care of advancing for fill?

Reply:
Actually it's not the case. `isElementwise(op)` returns true for `linalg.fill`.